src/components/robot/Client.ts

```diff
@@ -0,0 +1,234 @@
+import type {
+  PoseInFrame,
+  ResourceName,
+  Transform,
+} from '../../gen/common/v1/common_pb.esm';
+import Client from '../../Client';
+import type { Duration } from 'google-protobuf/google/protobuf/duration_pb';
+import type { Robot } from './Robot';
+import type { grpc } from '@improbable-eng/grpc-web';
+
+import { RobotServiceClient } from '../../gen/robot/v1/robot_pb_service.esm';
+import { promisify } from '../../utils';
+import proto from '../../gen/robot/v1/robot_pb.esm';
+
+interface ServiceHostInit {
+  serviceHost: string;
+  opts?: grpc.RpcOptions;
+}
+
+type RobotClientInit = Client | ServiceHostInit;
+
+const isClient = function (init: RobotClientInit): init is Client {
+  return init instanceof Client;
+};
+
+const isAddressInit = function (init: RobotClientInit): init is ServiceHostInit {
+  return Boolean((init as ServiceHostInit).serviceHost);
+};
+
+export class RobotClient implements Robot {
+  private client: RobotServiceClient;
+
+  constructor(init: RobotClientInit) {
+    if (isClient(init)) {
+      this.client = init.createServiceClient(RobotServiceClient);
+    } else if (isAddressInit(init)) {
+      this.client = new RobotServiceClient(init.serviceHost, init.opts);
+    } else {
+      throw new Error('invalid arguments');
+    }
+  }
+
+  private get robotService() {
+    return this.client;
+  }
+
+  // OPERATIONS
```

Comment:
[nit] These types of comments aren't really useful

Reply:
I'd prefer to keep them for now if that's ok - I found these cheap comment sections helpful in the python SDK.

maybe this is a sign that robot is becoming a bit of god class / api?

Reply:
I don't really think so. It's fairly small.